### ISSUE TITLE
bug: ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)

### ISSUE THREAD

**Issue by tullie** [localstack/localstack#11306]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I'm trying to upgrade from localstack/localstack-pro:2.3.2 to localstack/localstack-pro:3.6 but I'm seeing the following error when trying to deploy a CDK stack that contains a mysql db:

```
2024-08-03T21:50:44.844 DEBUG --- [et.reactor-3] l.p.c.s.rds.db_utils       : Determined and reserved port 4511 (requested port None) for RDS database
2024-08-03T21:50:44.893  INFO --- [functhread18] l.p.c.s.rds.engine_mysql   : Starting MySQL server on port 4511 - database "magnusdb", user "user"
2024-08-03T21:50:44.908 DEBUG --- [functhread18] l.u.c.container_client     : Getting networks for container: localstack-main
2024-08-03T21:50:44.910  WARN --- [functhread18] l.u.container_networking   : The specified 'MAIN_DOCKER_NETWORK' is not connected to the LocalStack container! Falling back to localstack
2024-08-03T21:50:44.910 DEBUG --- [functhread19] l.u.c.docker_sdk_client    : Creating container with attributes: {'self': <localstack.utils.container_utils.docker_sdk_client.SdkDockerClient object at 0x7f6f7c7f5
410>, 'image_name': 'mysql:8.0.35', 'name': 'ls-mysql-03b34fb9', 'entrypoint': None, 'remove': False, 'interactive': False, 'tty': False, 'detach': False, 'command': None, 'volumes': [VolumeBind(host_dir='/tmp/lo
calstack/tmp/rds/mysql/000000000000/instance/000000000000/us-east-2/magnus-db', container_dir='/var/lib/mysql', read_only=False)], 'ports': None, 'exposed_ports': [], 'env_vars': {'MYSQL_ROOT_PASSWORD': 'RsHmBu8E
Wi7SFQhh0OAURvz3GjpHJe0w', 'MYSQL_PASSWORD': 'RsHmBu8EWi7SFQhh0OAURvz3GjpHJe0w', 'MYSQL_USER': 'user', 'MYSQL_DATABASE': 'magnusdb'}, 'user': 'root', 'cap_add': None, 'cap_drop': None, 'security_opt': None, 'netw
ork': 'localstack', 'dns': None, 'additional_flags': None, 'workdir': None, 'privileged': False, 'labels': None, 'platform': None, 'ulimits': None, 'init': None, 'log_config': None}
2024-08-03T21:50:44.980 DEBUG --- [functhread19] l.u.c.docker_sdk_client    : Starting container d1bf723c295b03f9f5486556b7b8e33a637e95fd6bd8c1e351f891659097d241
2024-08-03T21:50:45.307 DEBUG --- [functhread19] l.u.c.container_client     : Getting ipv4 address for container d1bf723c295b03f9f5486556b7b8e33a637e95fd6bd8c1e351f891659097d241 in network localstack.
2024-08-03T21:50:45.312 DEBUG --- [functhread20] l.utils.server.tcp_proxy   : Starting TCP proxy bound on 0.0.0.0:4511 forwarding to 172.19.0.3:3306
2024-08-03T21:50:46.813  INFO --- [t.reactor-10] localstack.request.aws     : AWS cloudformation.DescribeStackEvents => 200
2024-08-03T21:50:46.911 DEBUG --- [functhread18] l.u.c.docker_sdk_client    : Executing command in container d1bf723c295b03f9f5486556b7b8e33a637e95fd6bd8c1e351f891659097d241: ['bash', '-c', 'mysqladmin ping -p$MY
SQL_ROOT_PASSWORD --protocol tcp']
2024-08-03T21:50:47.064 DEBUG --- [functhread18] l.u.c.docker_sdk_client    : Executing command in container d1bf723c295b03f9f5486556b7b8e33a637e95fd6bd8c1e351f891659097d241: ['bash', '-c', 'mysql -u root -p$MYSQ
L_ROOT_PASSWORD -e "grant all privileges on *.* to \'user\'@\'%\'" ']
2024-08-03T21:50:47.295  WARN --- [functhread18] l.p.c.s.rds.db_utils       : Unable to startup DB instance: ('Exec command returned with exit code 1', None, b"mysql: [Warning] Using a password on the command lin
e interface can be insecure.\nERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)\n") Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/pro/core/services/rds/db_utils.py.enc", line 63, in A
    try:B and B(status='creating');G=F.start_db_instance(*E,**D);B and B(status=_D);LOG.debug('DB started successfully.');return G
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/pro/core/services/rds/engine_mysql.py.enc", line 72, in start_db_instance
    B(f"grant all privileges on *.* to '{E}'@'%'");B('flush privileges');U();return O
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/pro/core/services/rds/engine_mysql.py.enc", line 62, in B
    def B(query):A=f'mysql -u root -p$MYSQL_ROOT_PASSWORD -e "{query}" ';assert(B:=D.container.exec(command=[P,'-c',A]));return B
                                                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/pro/core/utils/container/container.py.enc", line 21, in B
    return A(self,*B,**C)
           ^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/pro/core/utils/container/docker_container.py.enc", line 19, in exec
    def exec(self,command):return DOCKER_CLIENT.exec_in_container(container_name_or_id=self.id,command=command)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/utils/container_utils/docker_sdk_client.py", line 852, in exec_in_container
    raise ContainerException(
localstack.utils.container_utils.container_client.ContainerException: ('Exec command returned with exit code 1', None, b"mysql: [Warning] Using a password on the command line interface can be insecure.\nERROR 104
5 (28000): Access denied for user 'root'@'localhost' (using password: YES)\n")
 stdout: 'None' stderr: 'mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
'
```

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

Running with:
```
docker run localstack/localstack
```
 
# docker-compose.yml
```
services:
  localstack:
    container_name: "localstack-main"
    image: localstack/localstack-pro:3.6
    networks:
      - default
    ports:
      - "127.0.0.1:53:53" # only required for Pro (DNS)
      - "127.0.0.1:53:53/udp" # only required for Pro (DNS)
      - "443:443" # only required for Pro (LocalStack HTTPS Edge Proxy)
      - "4510-4559:4510-4559" # external service port range
      - "4566-4584:4566-4584" # LocalStack Edge Proxy

    environment:
      - DATA_DIR=${DATA_DIR-}
      - DEBUG=${DEBUG-1}
      - DISABLE_CORS_CHECKS=1
      - DOCKER_HOST=unix:///var/run/docker.sock
      - EC2_VM_MANAGER=docker
      - ECS_REMOVE_CONTAINERS=1
      - ENFORCE_IAM=0
      - LAMBDA_DOCKER_FLAGS=-v /var/run/docker.sock:/var/run/docker.sock
      - LAMBDA_DOCKER_NETWORK=host # lambdas need access to localstack & rds
      - LOCALSTACK_API_KEY=${LOCALSTACK_API_KEY-}
      - LOCALSTACK_HOST=localhost
      - LOG_LICENSE_ISSUES=${LOG_LICENSE_ISSUES-1}
      - LS_LOG=debug
      - NODE_TLS_REJECT_UNAUTHORIZED=0
      - RDS_MYSQL_DOCKER=1
      - MYSQL_IMAGE=mysql:8.0.35
      - SERVICES=apigateway,cloudformation,cloudwatch,ec2,iam,lambda,logs,s3,secretsmanager,sqs,ssm,sts,apigatewaymanagementapi,apigatewayv2,application-autoscaling,elbv2,elasticache,ecr,ecs,rds,cognito-idp,glue
    volumes:
      - "${LOCALSTACK_VOLUME_DIR:-/tmp/localstack}:/var/lib/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
    command: "sleep 100000000"
networks:
  default:
    name: localstack
    driver: bridge
```

Relevant CDK code:
```
        self._instance: rds.DatabaseInstance = rds.DatabaseInstance(
            self,
            "MagnusDBInstance",
            engine=mysql_engine,
            instance_type=ec2.InstanceType("t2.medium"),
            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PUBLIC),
            vpc=vpc.instance,
            parameter_group=parameter_group,
            enable_performance_insights=enable_performance_insights,
            multi_az=False,
            allocated_storage=allocated_storage,
            max_allocated_storage=max_allocated_storage,
            database_name="magnusdb",
            instance_identifier=instance_identifier,
            credentials=rds.Credentials.from_secret(secret),
            security_groups=[vpc.database_sg],
            publicly_accessible=publicly_accessible,
        )
```

### Environment

```markdown
OS: Ubuntu 20.04.3 LTS
LocalStack version: 3.6.1.dev20240725063846
LocalStack build date: 2024-07-25
LocalStack build git hash: 33f0744
```

### Anything else?

Noticed this issue: https://github.com/localstack/localstack/issues/7703 but doesn't seem to help and not exactly the same thing I believe.

**Comment by tullie** [comment]
Okay this looks like it was fixed by just making sure this was run before deployment:
```
sudo rm -rf /tmp/localstack
```

Would be curious if there's a way to avoid having to do this but it should unblock for now

**Comment by MarcelStranak** [comment]
Hi @tullie, as a start I would suggest to start with a clean docker-compose.yml file. You can find it at https://github.com/localstack/localstack/blob/master/docker-compose-pro.yml
With the newer versions a lot of new changes were made that could break your configuration.
You can find the release notes at https://discuss.localstack.cloud/c/announcement/5

To understand the `/tmp/localstack` folder please visit our documentation page https://docs.localstack.cloud/references/filesystem/#localstack-volume

**Comment by bentsku** [comment]
Hello @tullie and thanks again for the report!

I believe this particular issue has been fixed in `localstack/localstack-pro:3.7` and `latest`, we now properly clean up before starting the DB. Could you please give it a try without running the `rm` command? 

Thank you!
